pkg/importer/http-datasource.go

```diff
@@ -133,19 +133,13 @@ func (hs *HTTPDataSource) Info() (ProcessingPhase, error) {
 	if hs.contentType == cdiv1.DataVolumeArchive {
 		return ProcessingPhaseTransferDataDir, nil
 	}
-	if !hs.readers.Convert {
-		return ProcessingPhaseTransferDataFile, nil
-	}
 	if pullMethod, _ := util.ParseEnvVar(common.ImporterPullMethod, false); pullMethod == string(cdiv1.RegistryPullNode) {
```

Comment:
Planning to try getting rid of nbdkit here?  I don't think it's necessary

Reply:
Didn't this PR (https://github.com/kubevirt/containerized-data-importer/pull/2845) address some bug for using scratch space with pull node imports? @akalenyu wdyt?

Reply:
The major change in that PR is returning "convert" for registry node pull.  Can still do that.  I just don't think we need nbdkit

Reply:
I didn't have much time to test this today but I added a commit to get rid of nbdkit for pull node. Let's see how tests behave and I'll do more testing tomorrow.

Reply:
@mhenriks seems the version of qemu-img we use doesn't support running convert/info on http endpoints, that's why we need either scratch space or nbdkit. It works on my local version though (odd since I'm using an older version), but since we expect to backport this I think we should keep using nbdkit here.

Reply:
That does not sound correct to me.  What is the error you see?  Can you check the logs to see what args we are we passing to qemu-img?

Reply:
Just a simple `qemu-img info --output=json http://localhost:8100/tinyCore.iso` returns `qemu-img: Could not open 'http://localhost:8100/tinyCore.iso': Unknown protocol 'http'`. I've tried creating a debugging pod and manually running the command inside and it happens the same. However, running it locally works.

Reply:
Ah the container must not have `qemu-block-curl` package.

Since updating rpms in old branches is annoying.  We can hold off getting rid of nbdkit in this pr but I think we should get rid of it in a subsequent pr that is not backported